lnst/RecipeCommon/Perf/Measurements/LinuxPerfMeasurement.py

```diff
@@ -0,0 +1,240 @@
+        if result:
+            self._individual_results = [result]
+
+    @property
+    def individual_results(self) -> list[LinuxPerfMeasurementResults]:
+        return self._individual_results
+
+    def add_results(self, result: LinuxPerfMeasurementResults):
+        self._individual_results.append(result)
+
+
+class LinuxPerfMeasurement(BaseMeasurement):
+    _MEASUREMENT_VERSION: int = 1
+
+    hosts: list[Host]
+    intr_filename: str
+    intr_cpus: list[int]
+    iperf_filename: str
+    iperf_cpus: list[int]
+    _start_timestamp: float
+    _end_timestamp: float
+    _data_folder: str
+
+    _version: Optional[dict[str, Any]] = None
+    _collection_index: int = 0
+    _jobs: list[Job] = []
+
+    def __init__(
+        self,
+        hosts: list[Host],
+        intr_filename: str,
+        intr_cpus: list[int],
+        iperf_filename: str,
+        iperf_cpus: list[int],
+        data_folder: str,
+        recipe_conf: Any = None,
+    ):
+        super().__init__(recipe_conf)
+
+        self.hosts = hosts
+        self.intr_filename = intr_filename
+        self.intr_cpus = intr_cpus
+        self.iperf_filename = iperf_filename
+        self.iperf_cpus = iperf_cpus
+
+        self._data_folder = data_folder
+
+        # create output folders
+        for host in self.hosts:
+            try:
+                os.mkdir(os.path.join(self._data_folder, host.hostid))
+            except FileExistsError:
+                pass
+
+    @property
+    def version(self) -> Optional[dict[str, Any]]:
+        if self._version:
+            return self._version
+
+        perf_version: Optional[str] = self._get_host_perf_version()
+        if perf_version is None:
+            return None
+
+        self._version = {
+            "measurement_version": self._MEASUREMENT_VERSION,
+            "host_perf_version": perf_version,
+        }
+        return self._version
+
+    def _get_host_perf_version(self) -> Optional[str]:
+        if not len(self.hosts):
+            raise Exception(
+                "No hosts in LinuxPerfMeasurement while getting perf version"
+            )
+
+        version_job = self.hosts[0].run("perf --version", job_level=ResultLevel.DEBUG)
+        if version_job.passed:
+            match = re.match(r"perf version (.+?)", version_job.stdout)
+            if match:
+                return match.group(1)
+        return None
+
+    def _generate_configurations(self):
+        for host in self.hosts:
+            yield (host, self.intr_filename, self.intr_cpus)
+            yield (host, self.iperf_filename, self.iperf_cpus)
+
+    def start(self) -> None:
+        self._start_timestamp = time.time()
+        for host, filename, cpus in self._generate_configurations():
+            self._jobs.append(host.run(LinuxPerf(output_file=filename, cpus=cpus), bg=True))
+
+    def finish(self) -> None:
+        for job, (host, _, _) in zip(self._jobs, self._generate_configurations()):
```

Comment:
I'm a bit concerned about this and another call in `collect_results` of `_generate_configurations()`. Can't we instead save the job:configuration in the first call of that method on line 165 and reuse this here? Is there a reason why to do it this way? I understand that the functionality is the same, but it just looks weird to generate them again and again.

Reply:
Yeah, I think the property would look better.

Reply:
resolved in a00d00d